controller/space.go

```diff
@@ -216,23 +216,6 @@ func (c *SpaceController) Delete(ctx *app.DeleteSpaceContext) error {
 			ctx, errors.NewInternalError(ctx, errs.Wrapf(err, "failed to delete codebases associated with space %s", spaceID)))
 	}
 
-	// now delete the OpenShift resources associated with this space on an
-	// OpenShift cluster, unless otherwise specified
-	if ctx.SkipCluster == nil || !*ctx.SkipCluster {
-		err = deleteOpenShiftResource(c.DeploymentsClient, config, ctx.Context, spaceID)
```

Comment:
How are deployments deleted if not through the API?

Reply:
yeap, indeed another services will have to deal with deployment resources. 
This is the purpose of this PR remove deployment out of wit.
In an event-driven world, an event sent for delete space would triggered a subscribed deployment service.

Reply:
The `deleteOpenshiftResource` method fails a lot of times and we already have an issue to fix it https://github.com/openshiftio/openshift.io/issues/4657. Since @kwk will be implementing an improved version of `space deletion` (in future), I'm in favour of deleting this method.